runner/runners/queue.go

```diff
@@ -454,9 +454,13 @@ func (c *QueueController) runAndWatch(cmdID cmdAndID) chan runner.RunStatus {
 				}).Info("Queue received status update")
 			c.statusManager.Update(st)
 			if st.State.IsDone() {
-				if st.ExitCode == errors.CleanFailureExitCode && c.lastExitCode == errors.CleanFailureExitCode {
+				if st.ExitCode == int(c.lastExitCode) &&
+					(st.ExitCode == errors.CleanFailureExitCode ||
+						st.ExitCode == errors.CheckoutFailureExitCode ||
+						st.ExitCode == errors.GenericCheckoutFailureExitCode ||
+						st.ExitCode == errors.ReleaseCheckoutFailureExitCode) {
 					c.inv.stat.Counter(stats.WorkerServerKillGauge).Inc(1)
```

Comment:
do we want to sleep for 1 minute (the stats latch time) to give stats a chance to pick up this count?  This might tell us if a lot of workers are restarting due to errors.

Reply:
For now I'd recommend no, we would probably have to sleep longer to make sure obs had time to pick up the emitted metric to avoid flakiness. It would make sense to try to monitor all worker instance failures internally and not just from this cause if we were to spend more time on this.